### PR TITLE
Add categorical/multinomial distributions and simple Bernoulli model example

### PR DESCRIPTION
**Pull request by gizemcaylak**
Started adding simple probabilistic models, also added categorical/multinomial distributions to coreppl since I will need them for simple models. Simple Bernoulli model assumes n, 10, Bernoulli random variables with  probability 0.5.
Also, there is a .DS_Store file in coreppl which should be removed.

### CODE REVIEW

File: coreppl/dist.mc [hunk 1]
```diff
@@ -52,6 +52,32 @@ lang BetaDist = Dist
     else never
 end
 
+-- DCategorical {p=p} is equivalent to DMultinomial {n=1, p=p}
+lang CategoricalDist = Dist
+
+  syn Dist =
+  | DCategorical { p: Expr }
```
Comment (dlunde) [inline coreppl/dist.mc:59]: I'm not sure how explicit we should be with the arguments for distributions. For example `DBeta` now has two (explicit) arguments: `a` and `b`. We could equally well have a single expression `p` as argument for `DBeta` (similar to `DCategorical`), which we restrict to have type `{a: Float, b: Float}` in the type system.

Maybe the idea is to have explicit arguments only when there is a fixed number of arguments? For consistency, maybe we should just have all distributions take a single argument, and then handle everything in the type system.

File: coreppl/coreppl.mc [hunk 2]
```diff
@@ -127,7 +127,11 @@ utest expr2str (assume_ (bern_ (float_ 0.7)))
   with "assume (Bern 7.0e-1)" in
 utest expr2str (observe_ (float_ 1.5) (beta_ (float_ 1.0) (float_ 2.0)))
   with "observe 1.50e+0 (Beta 1.0e-0, 2.0e+0)" in
-utest expr2str (weight_ (float_ 1.5)) 
+utest expr2str (weight_ (float_ 1.5))
   with "weight 1.50e+0" in
+utest expr2str (assume_ (categorical_ (seq_ [float_ 0.3, float_ 0.2, float_ 0.5])))
+  with "assume (Categorical [ 3.0e-1,\n  2.0e-1,\n  5.0e-1 ])" in
+utest expr2str (assume_ (multinomial_ (int_ 5) (seq_ [float_ 0.3, float_ 0.2, float_ 0.5])))
+  with "assume (Multinomial 5, [ 3.0e-1,\n  2.0e-1,\n  5.0e-1 ])" in
```
Comment (dlunde) [inline coreppl/coreppl.mc:135]: These new utests should probably be moved to `dist.mc` instead (with `assume` removed)

Comment (gizemcaylak) [reply coreppl/coreppl.mc:135]: The problem with that is in `dist.mc`, sequences or const cannot be printed right now (same applies to `DBeta` and `DBernoulli`). Should I keep them in `coreppl.mc` or do you have any suggestion on that?

Comment (dlunde) [reply coreppl/coreppl.mc:135]: I did the following in my local branch (at the bottom of `dist.mc`):
```
-- Language compositions

lang DistAll = BernDist + BetaDist + ExpDist + EmpiricalDist

lang Test = DistAll + MExprAst + MExprPrettyPrint

mexpr

use Test in

utest expr2str (empirical_ [(1.0, float_ 1.5), (3.0, float_ 1.3)]) with "(Empirical size=2)" in
utest expr2str (exp_ (float_ 1.0)) with "(Exp 1.0e-0)" in

()

```

Maybe I should actually make a PR and merge that one first then?

Comment (dlunde) [reply coreppl/coreppl.mc:135]: Now added my PR #21.

**Comment by david-broman** [comment]
Gizem, will you change this to a PR (not WIP)?

**Pull request merged**